api/python/provisioner/commands/validator/validator.py

```diff
@@ -784,3 +787,128 @@ def __attrs_post_init__(self):
         self.content_validator = ContentFileValidator(
                 scheme=ThirdPartyReleaseInfoContentScheme,
                 content_type=self.content_type)
+
+
+@attr.s(auto_attribs=True)
+class CompatibilityValidator:
+
+    """
+    Validator that checks if all SW upgrade packages are compatible with
+    installed ones.
+    """
+
+    @staticmethod
+    def _compare_versions(a: str, b: str, operator: str) -> bool:
```

Comment:
that custom implemetation looks good, but I would suggest to simplify our life and use [packaging](https://packaging.pypa.io/en/latest/specifiers.html) here

```bash
>>> from packaging.specifiers import Version, SpecifierSet
>>> old_v = Version('1.0.0-321')
>>> new_v = Version('2.0.0-321')
>>> constraint = SpecifierSet('> 2.0.0-0')
>>> old_v in constraint
False
>>> new_v in constraint
True
>>> old_v2 = Version('2.0.0-0')
>>> old_v2 in constraint
False
```

Reply:
Yes, I rewrote this part using the proposed functionality, thank you! I didn't expect that `packaging` supports such API.